src/main/java/yapbot/util/Storage.java

```diff
@@ -0,0 +1,158 @@
+package yapbot.util;
+
+import yapbot.exceptions.YapBotException;
+import yapbot.tasks.Deadline;
+import yapbot.tasks.Event;
+import yapbot.tasks.Task;
+import yapbot.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Handles all file interactions for YapBot.
+ */
+public class Storage {
+    private File file;
+    private String filepath;
+
+    /**
+     * Returns a Storage instance.
+     * The actual file is not automatically created together with the instance.
+     *
+     * @param filepath Location where tasks are saved to.
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+        this.file = new File(filepath);
+    }
+
+    /**
+     * Parses the file to return the tasks stored on it.
+     * If the file is empty, an empty ArrayList is returned.
+     * If the file is corrupted or parsing fails, data on the file is dumped.
+     *
+     * @return An ArrayList of tasks parsed from the file.
+     * @throws YapBotException If the file cannot be parsed or the file cannot be created.
+     */
+    public ArrayList<Task> load() throws YapBotException {
+        try {
```

Comment:
I like how you handle the errors here but perhaps can make the code more organized by creating another method to control the error handling?